### PR TITLE
add default optInValue + expose current service status

### PR DESCRIPTION
**Pull request by draskovits**

### CODE REVIEW

File: GDPR/Utilities/String+Duplication.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
+//  File.swift
```
Comment (mpoimer) [inline GDPR/Utilities/String+Duplication.swift:2]: remove header by formatting file

File: GDPR/Utilities/String+Duplication.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+//
+//  Created by Stefan Draskovits on 29.11.22.
+//
+
+import Foundation
+
+extension Sequence where Element: Hashable {
+    var hasDuplicates: Bool {
+        var elements = Set<Element>()
+        for element in self {
```
Comment (mpoimer) [inline GDPR/Utilities/String+Duplication.swift:13]: couldn't we just put self into a Set and then compare the count of self against the count of the Set, and if the Set count is smaller, we know that there have to be duplicates?

File: GDPR/GDPRManager.swift [hunk 3]
```diff
@@ -67,50 +62,69 @@ public class GDPRManager {
...
         currentStatus = PersistenceManager.shared.retrieveStatus()
         self.termsURL = termsURL
         self.privacyPolicyURL = privacyPolicyURL
     }
-    
+
     public func setServices(services: [Service]) {
         guard let currentStatus = currentStatus else { return }
+        assert(!services.map { $0.id }.hasDuplicates, "The ids of the services have to be unique" )
         for service in services {
```
Comment (mpoimer) [inline GDPR/GDPRManager.swift:77]: I don't get this for loop... and how do I get rid of a service completely?

File: GDPR/GDPRManager.swift [hunk 4]
```diff
@@ -67,50 +62,69 @@ public class GDPRManager {
...
+
+    public func isServiceEnabled(serviceId: String) -> Bool? {
+        guard let currentStatus = currentStatus else { return nil }
+        return currentStatus
+            .services
+            .first { $0.id == serviceId }?
+            .isOptIn
+    }
     
+    public func fetchServiceStatus() -> [String: Bool]? {
```
Comment (mpoimer) [inline GDPR/GDPRManager.swift:97]: fetchServicesStatus

**Review by mpoimer — APPROVED** [review]

File: GDPR/GDPRManager.swift [hunk 5]
```diff
@@ -67,50 +62,69 @@ public class GDPRManager {
...
         guard let currentStatus = currentStatus else { return }
+        assert(!services.map { $0.id }.hasDuplicates, "The ids of the services have to be unique" )
         for service in services {
             for singleService in currentStatus.services where service.id == singleService.id {
                 singleService.description = service.description
                 singleService.name = service.name
                 singleService.supportDeletion = service.supportDeletion
                 return
             }
-            self.currentStatus?.services.append(ServiceModel(id: service.id, name: service.name, description: service.description, supportDeletion: service.supportDeletion, isOptIn: false))
```
Comment (mpoimer) [inline GDPR/GDPRManager.swift:88]: shouldn't isOptIn == false let the toggle be "on" by default?

File: GDPR/Utilities/String+Duplication.swift (hunk 2 above)
Comment (draskovits) [reply GDPR/Utilities/String+Duplication.swift:13]: Yes, your solution would work as well, but does it matter here? 

My solution has a better best case performance (e.g. [1,1,2,3....]) 
In worst case my solution is 2x calculating hash values, whereas your solution is 1x calculating hash values + 2x count.

File: GDPR/GDPRManager.swift (hunk 3 above)
Comment (draskovits) [reply GDPR/GDPRManager.swift:77]: this is nor part of the pr - we will have to refactor some stuff here

File: GDPR/GDPRManager.swift (hunk 5 above)
Comment (draskovits) [reply GDPR/GDPRManager.swift:88]: I don't understand the question

**Pull request merged**